src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3653 @@
+
+Let $b \in M$. Return $M$.
+"""
+parent(b::SubQuoElem) = b.parent
+
+@doc Markdown.doc"""
+    (R::SubQuo)(a::FreeModElem; check::Bool = true)
+
+Return `a` as an element of `R`. If `check == true` (default) it is checked that `a` represents
+indeed an element of `R`.
+"""
+function (R::SubQuo)(a::FreeModElem; check::Bool = true)
+  if check
+    b = convert(R.sum.gens.SF, a)
+    c = _reduce(b, singular_generators(std_basis(R.sum)))
+    iszero(c) || error("not in the module")
+  end
+  return SubQuoElem(a, R)
+end
+
+@doc Markdown.doc"""
+    (R::SubQuo)(a::SRow)
+
+Return the subquotient element $\sum_i R[i] \cdot a[i] \in R$.
+"""
+function (R::SubQuo)(a::SRow)
+  return SubQuoElem(a, R)
+end
+
+@doc Markdown.doc"""
+    (R::SubQuo)(a::SubQuoElem)
+
+Return `a` if it lives in `R`.
+"""
+function (R::SubQuo)(a::SubQuoElem)
+  if parent(a) == R
+    return a
+  end
+  error("illegal coercion")
+end
+
+@doc Markdown.doc"""
+    index_of_gen(v::SubQuoElem)
+
+Let $v \in G$ with $v$ the `i`th generator of $G$. Return `i`.
+"""
+function index_of_gen(v::SubQuoElem)
+  @assert length(coeffs(v).pos) == 1
+  @assert isone(coeffs(v).values[1])
+  return coeffs(v).pos[1]
+end
+
+# function to check whether a free module element is in a particular free module
+function check_parent(a::Union{FreeModElem,SubQuoElem}, b::Union{FreeModElem,SubQuoElem})
+  if parent(a) !== parent(b)
+    error("elements not compatible")
+  end  
+end
+
+function +(a::SubQuoElem, b::SubQuoElem)
+  check_parent(a,b)
+  return SubQuoElem(coeffs(a)+coeffs(b), a.parent)
+end
+function -(a::SubQuoElem, b::SubQuoElem) 
+  check_parent(a,b)
+  return SubQuoElem(coeffs(a)-coeffs(b), a.parent)
+end
+-(a::SubQuoElem) = SubQuoElem(-coeffs(a), a.parent)
+function *(a::MPolyElem_dec, b::SubQuoElem) 
+  if parent(a) !== base_ring(parent(b))
+    error("elements not compatible")
+  end
+  return SubQuoElem(a*coeffs(b), b.parent)
+end
+function *(a::MPolyElem, b::SubQuoElem) 
+  if parent(a) !== base_ring(parent(b))
+    error("elements not compatible")
+  end
+  return SubQuoElem(a*coeffs(b), b.parent)
+end
+function *(a::RingElem, b::SubQuoElem) 
+  if parent(a) !== base_ring(parent(b))
+    error("elements not compatible")
+  end
+  return SubQuoElem(a*coeffs(b), b.parent)
+end
+*(a::Int, b::SubQuoElem) = SubQuoElem(a*coeffs(b), b.parent)
+*(a::Integer, b::SubQuoElem) = SubQuoElem(a*coeffs(b), b.parent)
+*(a::fmpq, b::SubQuoElem) = SubQuoElem(a*coeffs(b), b.parent)
+==(a::SubQuoElem, b::SubQuoElem) = iszero(a-b)
+
+@doc Markdown.doc"""
+    sub(F::FreeMod, O::Vector{<:FreeModElem}, task::Symbol = :none)
+
+Return `S` as a submodule of `F`, where `S` is generated by `O`.
+`S` is a represented as a subquotient module.
+The elements of `O` must live in `F`.
+If `task` is set to `:with_morphism` or to `:both` return also the canonical injection morphism
+$S \to F$.
+If `task` is set to `:store` the morphism is also cached.
+If `task` is set to `:morphism` return only the morphism.
```

Comment:
This documentation does not match the code: on the one hand, `:with_morphism` is not actually handled by the code; on the other hand, the code also uses `:module` which is not mentioned here.